numba/compiler.py

```diff
@@ -559,29 +564,25 @@ def stage_parfor_pass(self):
             self.flags.auto_parallel, self.flags, self.parfor_diagnostics)
         parfor_pass.run()
 
-        if config.WARNINGS:
-            # check the parfor pass worked and warn if it didn't
-            has_parfor = False
-            for blk in self.func_ir.blocks.values():
-                for stmnt in blk.body:
-                    if isinstance(stmnt, Parfor):
-                        has_parfor = True
-                        break
-                else:
-                    continue
-                break
-
-            if not has_parfor:
-                # parfor calls the compiler chain again with a string
-                if not self.func_ir.loc.filename == '<string>':
-                    msg = ("parallel=True was specified but no transformation"
-                           " for parallel execution was possible.")
-                    warnings.warn_explicit(
-                        msg,
-                        errors.NumbaWarning,
-                        self.func_id.filename,
-                        self.func_id.firstlineno
-                        )
+        # check the parfor pass worked and warn if it didn't
+        has_parfor = False
+        for blk in self.func_ir.blocks.values():
+            for stmnt in blk.body:
+                if isinstance(stmnt, Parfor):
+                    has_parfor = True
+                    break
+            else:
+                continue
+            break
+
+        if not has_parfor:
+            # parfor calls the compiler chain again with a string
+            if not self.func_ir.loc.filename == '<string>':
+                msg = ("\nThe keyword argument 'parallel=True' was specified "
+                       "but no transformation for parallel execution was "
+                       "possible.\n\nTo find out why, try turning on parallel "
+                       "diagnostics, see URL for help.")
+                warnings.warn(errors.NumbaWarning(msg, self.func_ir.loc))
```

Comment:
Seems like this could be an instance of `NumbaPerformanceWarning`, since the user is likely turning on `parallel=True` to improve performance.

Reply:
Agree. Also, URL needs to be an actual URL not just the letters `'URL'`, will fix!